src/Helper/Helper.php

```diff
@@ -68,18 +93,198 @@ public function getSubmissionIdFromJob(int $jobId): ?int {
   }
 
   /**
-   * Get all jobs from advanced queue table.
+   * Get all jobs that match a specific form.
+   *
+   * @param string $formId
+   *   The form to match.
    *
    * @return array
-   *   A list of jobs.
+   *   A list of view parameters.
    *
-   * @phpstan-return array<int, \stdClass>
+   * @phpstan-return array<string, mixed>
    */
-  public function getAllJobs():array {
-    $query = $this->connection->select('advancedqueue', 'a');
-    $query->fields('a', ['payload', 'job_id']);
+  public function getQueueJobIds(string $formId): array {
+    $query = $this->connection->select('os2forms_queue_submission_relation', 'o');
+    $query->fields('o', ['job_id']);
+    $query->condition('webform_id', $formId, '=');
+
+    return $query->execute()->fetchCol();
+  }
+
+  /**
+   * Handle a job from the AdvancedQueueProcessSubscriber.
+   *
+   * @param \Drupal\advancedqueue\Job $job
+   *   The job about to be processed.
+   */
+  public function handleJob(Job $job): void {
+    $data = $this->getDataFromPayload($job->getPayload());
+    if ($data) {
+      $data['job_id'] = (int) $job->getId();
+      $this->addRelation($data);
+    }
+  }
+
+  /**
+   * Handle importing advanced queue jobs through a command.
+   */
+  public function handleImport(): void {
+    $jobs = $this->getAllQueueJobs();
+    foreach ($jobs as $job) {
+      $this->handleJob($job);
+    }
+  }
+
+  /**
+   * Get webform id from job queue id.
+   *
+   * @param string $jobId
+   *   The id of a queue job.
+   *
+   * @return string
+   *   The webform id.
+   */
+  public function getWebformIdFromQueue(string $jobId): string {
+    $query = $this->connection->select('os2forms_queue_submission_relation', 'o');
+    $query->condition('job_id', $jobId, '=');
+    $query->fields('o', ['webform_id']);
+
+    return $query->execute()->fetchObject()->webform_id;
```

Comment:
You should handle the case when the requested job id does not exist.

Reply:
Fixed using null safe operator